banidb/banidb.py

```diff
@@ -0,0 +1,409 @@
+    p=[]
+    for i in r['index']:
+        l=[i['ShabadID'],i['GurmukhiUni'],i['Translations']['en']['bdb'],i['SourceUnicode']]
+        p.append(l)
+    d['Banis']=p
+    return d
+
+def amritkeertanshabad(shabadID):
+    x=shabad(shabadID)
+    return x
+
+def kosh(letter):
+    link='http://api.banidb.com/v2/kosh/'+str(letter)
+    q=requests.get(link)
+    r=q.json()
+    d=[['ਸ਼ਬਦ','Word']]
+    for i in range(len(r)):
+        f=r[i]
+        d.append([f['wordUni'],f['word']])
+    return d
+        
+
+def koshword(word):
+    link='http://api.banidb.com/v2/kosh/word/'+str(word)
+    q=requests.get(link)
+    r=q.json()
+    d=[['ਸ਼ਬਦ (Word)','ਵਿਆਖਿਆ (Definition)']]
+    for i in r:
+        d.append([i['wordUni'],i['definitionUni']])
+    return d
+
+def koshsearch(query):
+    link='http://api.banidb.com/v2/kosh/search/'+str(query)
+    q=requests.get(link)
+    r=q.json()
+    d=[['ਸ਼ਬਦ (Word)','ਵਿਆਖਿਆ (Definition)']]
+    for i in range(len(r)):
+        f=r[i]
+        d.append([f['wordUni'],f['definitionUni']])
+    return d
+
+def rehats():
+    link='http://api.banidb.com/v2/rehats'
+    q=requests.get(link)
+    r=q.json()
+    d=[['Rehat ID','Rehat Name']]
+    for i in r['maryadas']:
+        d.append([i['rehatID'],i['rehatName']])
+    return d
+        
+
+def rehat(rehatID):
+    link='http://api.banidb.com/v2/rehats/'+str(rehatID)
+    q=requests.get(link)
+    r=q.json()
+    d={'Rehat ID':rehatID,'chapters':[]}
+    for i in r['chapters']:
+        l={'Chapter ID':i['chapterID'],'Chapter Name':i['chapterName']}
+        d['chapters'].append(l)
+    return d            
+    
+def rehatChapter(rehatID,chapterID):
+    link='http://api.banidb.com/v2/rehats/'+str(rehatID)+'/chapters/'+str(chapterID)
+    q=requests.get(link)
+    d=q.json()
+    return d
+
+def rehatSearch(query): #remove extra html tags from API
+    link='http://api.banidb.com/v2/rehats/search/'+str(query)
+    q=requests.get(link)
+    r=q.json()
+    d=[]
+    for i in r['rows']:
+        l={'Rehat ID':i['rehatID'],'Chapter ID':i['chapterID'],'Chapter Name':i['chapterName']}
+        d.append(l)
+    return d
+
+def writers():
+    link='http://api.banidb.com/v2/writers'
+    q=requests.get(link)
+    r=q.json()
+    d=[]
+    for i in r['rows'][1:]:
+        l={'WriterID':i['WriterID'],'WriterName':i['WriterEnglish']}
+        d.append(l)
+    return d
+
+#Updated on 21/04/2021 - 16:16 IST
+def raags():
+    link='http://api.banidb.com/v2/raags'
+    q=requests.get(link)
+    r=q.json()
+    d=[]
+    for i in r['rows'][1:]:
+        l={
+            'Raag ID':i['RaagID'],
+            'ਰਾਗ':i['RaagUnicode'],
+            'Raag':i['RaagEnglish']
+            }
+        d.append(l)
+    return d
+
+def raag(RaagID):
+    link='http://api.banidb.com/v2/raags'
+    q=requests.get(link)
+    r=q.json()
+    d=[]
+    for i in r['rows'][1:]:
+        if i['RaagID']==RaagID:
+            linkw='http://api.banidb.com/v2/writers'

```

Comment:
Maybe refactor the base of the url  `https://api.bandidb.com` and version `v2` as global constants and construct the links within each method - that way in future it will be a single line change, should either of these change.

Also, let's use `https://` instead of `http://`

Reply:
Done